circuit_knitting_toolbox/circuit_cutting/cutting_decomposition.py

```diff
@@ -83,6 +92,132 @@ def partition_circuit_qubits(
+    """
+    # Replace specified gates with TwoQubitQPDGates
+    if not inplace:
+        circuit = circuit.copy()
+    for gate_id in gate_ids:
+        gate = circuit.data[gate_id]
+        qubit_indices = [circuit.find_bit(qubit).index for qubit in gate.qubits]
+        decomposition = QPDBasis.from_gate(gate.operation)
+        qpd_gate = TwoQubitQPDGate(decomposition, label=f"cut_{gate.operation.name}")
+        circuit.data[gate_id] = CircuitInstruction(
+            qpd_gate, qubits=qubit_indices
+        )
+
+    return circuit
+
+
+def partition_problem(
+    circuit: QuantumCircuit,
+    partition_labels: Sequence[str | int],
+    observables: PauliList | None = None,
+) -> tuple[dict[str | int, QuantumCircuit], dict[str | int, PauliList] | None]:
+    """
+    Separate an input circuit and observable(s) along qubit partition labels.
+
+    Circuit qubits with matching partition labels will be grouped together, and non-local
+    operations spanning more than one partition will be decomposed and replaced with
+    probabilistic local operations.
+
+    If provided, the observables will be separated along the boundaries specified by
+    ``partition_labels``.
+
+    Args:
+        circuit: The circuit to separate
+        partition_labels: A sequence of labels, such that each label corresponds
+            to the circuit qubit with the same index
+        observables: The observables to separate
 
-    return new_qc
+    Returns:
+        A tuple containing a dictionary mapping a partition label to a subcircuit
+        and a dictionary mapping a partition label to a list of Pauli observables
+
+    Raises:
+        ValueError: The number of partition labels does not equal the number of qubits in the circuit.
+        ValueError: An input observable acts on a different number of qubits than the input circuit.
+    """
+    if len(partition_labels) != circuit.num_qubits:
+        raise ValueError(
+            f"The number of partition labels ({len(partition_labels)}) must equal the number "
+            f"of qubits in the circuit ({circuit.num_qubits})."
+        )
+    if (
+        observables is not None
+        and len([obs for obs in observables if len(obs) != circuit.num_qubits]) != 0
+    ):
+        raise ValueError(
+            "An input observable acts on a different number of qubits than the input circuit."
+        )
+
+    # Partition the circuit with TwoQubitQPDGates and assign the order via their labels
+    qpd_circuit = partition_circuit_qubits(circuit, partition_labels)
+    i = 0
+    for inst in qpd_circuit.data:
+        if isinstance(inst.operation, TwoQubitQPDGate):
+            inst.operation.label = inst.operation.label + f"_{i}"
+            i += 1
+
+    # Separate the decomposed circuit into its subcircuits
+    qpd_circuit_dx = qpd_circuit.decompose(TwoQubitQPDGate)
+    separated_circs = separate_circuit(qpd_circuit_dx)
```

Comment:
```suggestion
    separated_circs = separate_circuit(qpd_circuit_dx, partition_labels)
```
otherwise the labels get their default: 0, 1, ...
and with this change, `subcircuits` and `subobservables` will have the same keys in the tutorial.